src/python/pants/option/global_options.py

```diff
@@ -615,6 +623,15 @@ def register_bootstrap_options(cls, register):
             advanced=True,
             help="Number of concurrent processes that may be executed remotely.",
         )
+        register(
+            "--process-execution-cache-namespace",
+            advanced=True,
```

Comment:
Defaults to None?

I also like to set `type=str`. Makes it more explicit to readers of this code.